Date/Date.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include "Date.h"
+
+using namespace std;
+
+#define FEBRUARY 1
+#define MONTH_AMOUNT 12
+
+void Date::validate(int day, int month, int year) {
+    int daysInMonth[MONTH_AMOUNT] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+    int indexMonth = month - 1;
+
+    if ( year % 400 == 0 || year % 4 == 0 && year % 100 != 0 ) {
+        daysInMonth[FEBRUARY] += 1;
+    }
+
+    if ( day > daysInMonth[indexMonth] || day < 1 ) {
+        throw InvalidDateException("Invalid Day!");
+    }
+
+    if ( month < 1 || month > MONTH_AMOUNT ) {
+        throw InvalidDateException("Invalid Month!");
+    }
+}
+
+Date::Date(int day, int month, int year) {
+    this->day = day;
+    this->month = month;
+    this->year = year;
+
+    validate(day, month, year);
```

Comment:
My logic tells me that validation should be before values assigning to object fields